---
layout: default
title: MeLanGE description
nav_order: 1
---

# MeLanGE Description

## The problem
Machine learning is an area of artificial intelligence that is gaining popularity in all fields of knowledge, including modern research in the life sciences. However, its use in comparative genomics of bacteria, especially as a tool in microbiome studies, is still in its infancy. This is mainly because there are no easy-to-use tools that meet the needs of researchers. With the advent of high-throughput DNA sequencing technologies, the amount of genomic data available far exceeds the amount of data that is thoroughly analysed. This can be partly explained by the difficulties of sorting and comparing large amounts of data, which is usually computationally intensive and often intractable. However, such large comparative genomics studies can be essential to determine important genomic or functional traits of different groups of organisms based on phylogeny, taxonomy, ecosystem origin, etc.

## Our solution
**MeLanGe** (**Machine Learning for Genomics**) aims to facilitate large-scale comparative genomic studies by combining different annotation schemes and machine learning techniques to rapidly detect characteristic features between genome/metagenome groups in highly complex datasets. MeLanGe performs automatic, multi-parallel genome annotations using various databases such as Pfam, COG, Kegg and CAZYmes and provides these annotations in a tabular format that can be used in further studies. In addition, MeLanGe can perform a semi-automatic feature selection process based on user-entered metadata to determine which key features better characterise each genome group.

## Performance
In a pilot study, out of 6986 Pfam features identified in 1256 marine and terrestrial bacterial genomes of the family _Flavobacteriaceae_, MeLanGe was able to reduce the complexity of the dataset to only 81 features that show a high correlation with the origin of the organism and thus serve as indicators of environmental specialisation. In summary, this tool can be an important player in the transition of microbiome studies into the "Big Data" era and offers an unprecedented opportunity to easily explore large groups of genomes.


* * *

